test/net.test.js

```diff
@@ -1,184 +1,159 @@
+    let insurancesCalcStub;
+    let taxCalcStub;
+
+    let net;
+
+    beforeEach(function () {
+        insurancesCalcStub = sinon.stub();
+        taxCalcStub = sinon.stub()
+
+        net = proxyquire("../src/net.js", {
+            "./insurancesCalculator.js": insurancesCalcStub,
+            "./taxCalculator.js": taxCalcStub,
+        });
+    });
+
+    afterEach(function () {
+        sinon.restore();
+    })
 
     describe('Check Dependent deduction', function () {
         //Personal Income Tax (PIT)
+        const mockedInsurance = { total: 0, insurances: [] };
+        const mockedTaxes = { totalTax: 0, rates: [] };
+        beforeEach(function () {
+            insurancesCalcStub.returns(Promise.resolve(mockedInsurance));
+            // insurancesCalcStub.resolves(mockedInsurance);
+
+            taxCalcStub.returns(
+                new Promise(function (resolve, reject) {
+                    resolve(mockedTaxes);
+                })
+            )
+        })
+
+        afterEach(function () {
+            sinon.restore();
+        })
 
-        test('No dependent deducted', function () {
+        test('No dependent deducted', async function () {
             //Gross income below 11 million is tax-exempt.
-            const actualPayslip = net(12_290_500, 0);
+            const gross = 12_290_500;
 
             const expectedPayslip = {
-                gross: 12_290_500,
+                gross: gross,
                 dependents: 0,
                 dependentDeductionAmount: 0,
                 region: 1,
-                insurances: [
-                    {
-                        amount: 983_240,
-                        name: 'Social Insurance 8%'
-                    },
-                    {
-                        amount: 184357.5,
-                        name: 'Health Insurance 1.5%'
-                    },
-                    {
-                        amount: 122905,
-                        name: 'Unemployment Insurance 1%'
-                    },
-                ],
-                totalInsurance: 1290502.5,
-                afterInsurance: 10999997.5,
-                taxes: [],
-                totalTax: 0,
-                netSalary: 10999997.5
+                insurances: mockedInsurance.insurances,
+                totalInsurance: mockedInsurance.total,
+                afterInsurance: gross - mockedInsurance.total,
+                taxes: mockedTaxes.rates,
+                totalTax: mockedTaxes.totalTax,
+                netSalary: gross - mockedTaxes.totalTax - mockedInsurance.total
             };
-            assert.deepStrictEqual(actualPayslip, expectedPayslip);
-        })
 
-        test('Dependent with no taxable income', function () {
-            //afterInsurance below 11 million is tax-exempt.
+            await net(gross).then(function (result) {
 
-            const actualPayslip = net(12_290_500, 1);
+                assert(taxCalcStub.calledWith(1290500));
```

Comment:
move it outside then (after await). The reasons:
- net.then is a function to get the actual payslip to continue next statements.
- taxCalcStub has nothing to do inside then.